src/schema/posts.ts

```diff
@@ -2768,29 +2768,67 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
       _,
       { postId }: { postId: string },
       ctx: AuthContext,
-    ): Promise<GQLEmptyResponse> => {
+    ): Promise<TransactionCreated> => {
+      const { userId } = ctx;
       const post = await validatePostBoostPermissions(ctx, postId);
       const campaignId = post?.flags?.campaignId;
 
       if (!campaignId) {
         throw new ValidationError('Post is not currently boosted');
       }
 
-      await ctx.con.transaction(async (entityManager) => {
+      const result = await ctx.con.transaction(async (entityManager) => {
         await entityManager
           .getRepository(Post)
           .update(
             { id: postId },
             { flags: updateFlagsStatement<Post>({ campaignId: null }) },
           );
 
-        await skadiApiClient.cancelPostCampaign({
+        const { currentBudget } = await skadiApiClient.cancelPostCampaign({
           campaignId,
           userId: ctx.userId,
         });
+        const toRefund = parseFloat(currentBudget);
+
+        const userTransaction = await entityManager
+          .getRepository(UserTransaction)
+          .save(
+            entityManager.getRepository(UserTransaction).create({
+              id: randomUUID(),
+              processor: UserTransactionProcessor.Njord,
+              receiverId: userId,
+              status: UserTransactionStatus.Success,
+              productId: null,
+              senderId: systemUser.id,
+              value: usdToCores(toRefund),
+              valueIncFees: 0,
+              fee: 0,
+              flags: { note: 'Post boost canceled' },
+              referenceId: campaignId,
+              referenceType: UserTransactionType.PostBoost,
+            }),
+          );
+
+        const transfer = await transferCores({
```

Comment:
try catch missing here with `TransferError` so that its handled correctly.

Reply:
what happens if cores refund fails on njord for any reason, should skadi campaign stay cancelled?

Reply:
Yes, it should stay cancelled. We'll have to do some intervention if that happens.

Reply:
do you want to log something specific and setup bew critical alert over those logs?